lib/facter/resolvers/solaris/dmi.rb

```diff
@@ -37,8 +37,10 @@ def read_facts(fact_name)
 
             output = exec_smbios(param[0])
             facts = param[1]
-            facts.each do |name, regx|
-              @fact_list[name] = output.match(/#{regx}/)&.captures&.first
+            if output
```

Comment:
This could be replaced with `return unless output`

Reply:
done